Edit Submariner's README

dfarrell07 opened the pull request:
Various edits to clean up Submariner's README.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

dfarrell07 committed:
Edit Submariner's README

Various edits to clean up Submariner's README.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

nyechiel commented on README.md:
````diff
@@ -64,31 +63,25 @@ experience and additionally provides testing and bug-diagnosing capabilities.
-   oc adm policy add-scc-to-user privileged system:serviceaccount:submariner:submariner-routeagent
-   oc adm policy add-scc-to-user privileged system:serviceaccount:submariner:submariner-gateway
-   ```
+See the [Known Issues docs](https://submariner.io/operations/known-issues/) on Submariner's website.
````
The SCC issue mentioned above (lines 85-90) is not documented under https://submariner.io/operations/known-issues/. Let's add it there before we remove it from this README. Also, AFAICT, we already take care of this with `subctl` as well as `submariner-addon` driven deployments, so I _think_ that users only need to do this for Helm. @skitt can you confirm?

skitt replied:
That’s correct, we only need this for Helm (and we could take care of it there too).

dfarrell07 replied:
https://github.com/submariner-io/submariner-website/pull/640

dfarrell07 replied:
https://github.com/submariner-io/submariner-charts/issues/195

dfarrell07 commented on README.md:
```diff
@@ -64,31 +63,25 @@ experience and additionally provides testing and bug-diagnosing capabilities.
 
-See the [`subctl verify` docs on Submariner's website](https://submariner.io/operations/deployment/subctl/#verify).
+See the [`subctl verify` docs](https://submariner.io/operations/deployment/subctl/#verify) and [Automated
+Troubleshooting docs](https://submariner.io/operations/troubleshooting/#automated-troubleshooting) on Submariner's website
```
```suggestion
Troubleshooting docs](https://submariner.io/operations/troubleshooting/#automated-troubleshooting) on Submariner's website.
```

nyechiel commented:
@dfarrell07 while on it, I also wanted to propose two additional changes (I am writing this as a comment and not as a code review feedback, because you haven't modified these lines):

- [Line 26] "Submariner is a tool built to connect overlay networks of different Kubernetes clusters... Submariner should be compatible with **any** CNI cluster network provider" --> I think we should say that "Submariner should be compatible with **most** CNI cluster network provider

- [Line 30] "Note that Submariner is in the **pre-alpha** stage, and should not be used for production purposes. While we welcome usage and experimentation, it is quite possible that you could run into bugs"  --> I am not sure what would the best wording to use ("early stage"? "evolving"?), but I feel like we past the pre-alpha stage

nyechiel commented on README.md:
```diff
@@ -14,28 +14,27 @@
 provider, as it utilizes off-the-shelf components to establish encrypted tunnels between each Kubernetes cluster.
 
-Note that Submariner is in the **pre-alpha** stage, and should not be used for production purposes. While we welcome usage and
+Note that Submariner is in an early stage, and should not be used for production purposes. While we welcome usage and
```
```suggestion
Note that Submariner is in an early stage, and while we welcome usage and
```

nyechiel replied:
Folks are already using it in production, so let's remove this wording.

dfarrell07 replied:
Yeah, but should they be? I thought many of the goals of 0.12 were to get SubM ready for production, which would imply we don't think it's ready for production now.

nyechiel replied:
@skitt @sridhargaddam wdyt?

skitt replied:
What does “ready for production” mean for an open source project anyway? Submariner as it exists now works and is usable in many scenarios.

0.12 goals are more about supportability, _i.e._ our ability to help people who are running into difficulties using Submariner. There isn’t much going into 0.12 which improves the core features of Submariner itself in a significant way.

dfarrell07 replied:
I guess in one sense “ready for production” means we would recommend running it in production. Our work on supportability seems to be a prerequisite for us "officially" making that recommendation. Others prerequisites we've identified that come to mind include rotating certs and an automated/reliable uninstall flow.

dfarrell07 replied:
Maybe someone else (who feels more comfortable with that change) would like to send a PR for this, and it can be discussed independently?

dfarrell07 replied:
I was trying to craft some caution middle-ground, but maybe it is best just to say nothing about production and make it clear in the commit message that this is not an endorsement of running in production.

dfarrell07 replied:
I sent a second commit dedicated to this change, and explained the reasoning in the commit message.

skitt replied:
Right, that matches my thoughts — there’s a difference between explicitly discouraging production use, explicitly encouraging it, and somewhere in between.

dfarrell07 committed:
Remove guidance about SubM in production

Don't provide guidance one way or the other about running Submariner in
production. This is not an endorsement of running the current version of
Submariner in production, but an acknowledgement that "production-ready"
means many different things to different people and we shouldn't try to
decide that for them.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

sridhargaddam commented on README.md:
```diff
@@ -14,28 +14,27 @@
 
 Submariner is a tool built to connect overlay networks of different Kubernetes clusters. While most testing is performed against Kubernetes
-clusters that have enabled Flannel/Calico/Canal/Weave/OpenShiftSDN, Submariner should be compatible with any CNI cluster network
+clusters that have enabled Flannel/Calico/Canal/Weave/OpenShiftSDN, Submariner should be compatible with most CNI cluster network
 provider, as it utilizes off-the-shelf components to establish encrypted tunnels between each Kubernetes cluster.
```
```suggestion
clusters that have enabled Flannel/Calico/Canal/Weave/OpenShiftSDN, Submariner should be compatible with most CNI cluster network
providers, as it utilizes off-the-shelf components to establish encrypted tunnels between each Kubernetes cluster.
```
